### ISSUE TITLE
Encounter training issues when retargeting HUMOTO dataset by MANIPTRANS

### ISSUE THREAD

**Issue by shengyin1224** [ManipTrans/ManipTrans#57]
Hi, Thanks for your wonderful work! We have carefully looked into codes and your paper, find that though HUMOTO dataset uses ADOBE Mixamo character model not SMPLX but provide all data needy by Maniptrans. So we decide to retarget this dataset and follow your guidance in README. 

The first Preprocessing step seems to be normal but the second training step has problems:

1. the object and the hand fastly move and the episode close just in 2s, we try to use COACD to the object assets but it seems not to be able to solve this problem.
2. we train 300 epochs by 2048 num envs, the training time by one epoch is too long (maybe 30 seconds) and the reward is almost not up (just from 15 to 18, I find OkaInk's reward has more than 100)

Here is the video. I am very grateful if you can help me about the reason and if I maybe forget some codes about SMPLX which not suit for Mxiamo? Maybe ADOBE Mixamo just can not be retargetted by Maniptrans? Thanks for your reply!

https://github.com/user-attachments/assets/443217c1-77d0-4026-b5e8-9c07aab3a510

**Comment by KailinLi** [comment]
Hi, thank you very much for your interest in ManipTrans and for taking the time to adapt it to a new dataset.

First of all, as long as the data provides all the signals required by ManipTrans (e.g., fingertip positions, wrist orientations, etc.), we believe it should in principle be possible to transfer to your dexterous hand setup, even if the characters are from ADOBE Mixamo instead of SMPLX. That said, given that the motion patterns are different, a few additional checks may be necessary:
1.	Frame rate:
Please make sure the motion data is at 60 Hz. You may need to interpolate or downsample your data to match the current training setting.
2.	Preprocessing / retargeting sanity check:
During the preprocessing (retargeting) stage, it’s very important to visualize the results and confirm that the hand–object poses are reasonable (e.g., no severe mesh penetration, no large gaps between hand and object when there should be contact).
3.	failed_execute thresholds:
You might consider relaxing the failed_execute criteria, especially for more difficult motions that are inherently harder to track. Strict thresholds can cause episodes to terminate very early and make training unstable.
4.	World frame and table alignment:
From the video, it looks like many interactions are happening relatively far away from the table surface. Please double-check whether the motions in your dataset are really intended to happen in free space, or if they should be closer to the table.
If the original dataset assumes tabletop manipulation, you may need to adjust the world-to-simulation transform so that the table, hand, and object in simulation are properly aligned with the dataset’s coordinate system.

In addition, some previous issues may contain related discussions and common pitfalls that could be helpful:
1.	https://github.com/ManipTrans/ManipTrans/issues/22#issuecomment-2826796932
2.	https://github.com/ManipTrans/ManipTrans/issues/47#issue-3499853639
3.	https://github.com/ManipTrans/ManipTrans/issues/54#issue-3616056920
4.	https://github.com/ManipTrans/ManipTrans/issues/55#issue-3640803905

You might find some similar symptoms and potential solutions there.

**Comment by shengyin1224** [comment]
Thanks for your patient reply! I have solved some bugs and problems, but now I have a new question:

The humoto dataset contains many tasks involving more than 1 object, if there are four objects in one motion, can I use Maniptrans to train? I have already modifed some codes to make the observation and reward supporting multiple objects but still do not know if it is feasible.

**Comment by shengyin1224** [comment]
<img width="791" height="461" alt="Image" src="https://github.com/user-attachments/assets/213df7cd-bf8a-495f-8637-78ac097719c3" />

I have another question: when I use mano2dexhand.py, I find that the DexHand may be occluded by objects, causing the hand to be unable to correctly optimize to the appropriate position in the visual representation. Is this normal? Do I need to modify the code to allow it to penetrate objects?

**Comment by KailinLi** [comment]
Hi, thanks a lot for the detailed update and for pushing this further.

For your first question: we haven’t actually tried training ManipTrans on scenarios with multiple interactive objects, but in principle it should be feasible. The key is that in the code you need to clearly distinguish between **"the object currently being manipulated"** and "all other objects". Since the state of the manipulated object is part of the policy observation, you may need to add some logic for switching which object is considered "active" and make sure the observation and reward are always consistent with that choice.
If most of the objects in the scene are static, then the problem becomes much simpler. You can treat them more like background geometry. For reference, you might look at the `FAVOR` data in `DexManipNet`, where there are many static tabletop geometries.

Regarding your second question: yes, the issue you described does need some extra handling. If the initial hand pose is in a very abnormal or heavily occluded configuration, it can indeed make optimization and policy learning much harder to converge. One possible solution is to also distinguish whether an object is currently being manipulated inside `mano2dexhand.py`. For objects that are not manipulated in the current clip, you might consider not spawning them at all, or disabling their collision shapes (e.g., via the corresponding flags in `gym.create_actor`). This can reduce occlusions and unwanted collisions for the DexHand.

**Comment by shengyin1224** [comment]
Thanks for your reply! Actually, regarding the second question: the handled bowl shown in the images is indeed an object that needs to be manipulated, but the hand cannot converge to the correct position by pure optimization (it should curl the fingers around the handle to grasp it). At this point I can

1. Disable the collision shape of this manipulated object? I have tried that, but the hand still can’t curl around the handle — it ends up with an outstretched palm.
2. Or add a touch/contact penalty term to the loss? But I feel that approach might be rather complicated.

As you mentioned earlier, I suspect the problem may lie in the Phase 1 optimization, or it could simply be that the tasks themselves are quite difficult (they require holding a bowl with one hand and a spoon with the other, plus other cabinet-moving–type tasks). Training in Phase 2 has been very challenging for me. Currently I’ve tried increasing the failure thresholds for both objects and hands by sixfold (otherwise training is nearly impossible), and I implemented adaptive curriculum learning (increase difficulty if success rate stays above 20% for five consecutive epochs; decrease difficulty if it fails to increase for ten consecutive epochs). Even so, after 10k training iterations it is still far from an acceptable state.

<img width="775" height="475" alt="Image" src="https://github.com/user-attachments/assets/2a3500c3-3859-4e71-89c1-79633fb18d02" />

<img width="775" height="472" alt="Image" src="https://github.com/user-attachments/assets/69d1eb6a-a83b-486b-b9a3-ed8c6496b642" />

<img width="783" height="473" alt="Image" src="https://github.com/user-attachments/assets/f9dc7a7d-0e66-4347-b2dd-0138f3ca99d2" />

**Comment by KailinLi** [comment]
We discussed this privately on WeChat and the issue is resolved. I’m leaving a few tips here for others who may run into a similar problem:

1.	Initialize the fitting based on the hand–object interaction location, rather than the object’s geometric center. This is crucial for objects with handles.
2.	Slightly adjust the dexhand position along the surface normal of the interaction region on the object mesh. This can effectively avoid severe finger–object penetration at the early stage of optimization.
3.	During retargeting, do not enable collision checking with the table, as it may hinder finding a better global optimum.

Video below shows my results.

https://github.com/user-attachments/assets/07d2b6fc-6576-48cd-88c2-5abcfd31f161

Closing this for now — happy to reopen if new questions arise.
